dlt/extract/extract.py

```diff
@@ -121,23 +126,38 @@ def _write_static_table(resource: DltResource, table_name: str) -> None:
                 # TODO: many resources may be returned. if that happens the item meta must be present with table name and this name must match one of resources
                 # if meta contains table name
                 resource = source.resources.find_by_pipe(pipe_item.pipe)
+                table_name: str = None
                 if isinstance(pipe_item.meta, TableNameMeta):
                     table_name = pipe_item.meta.table_name
                     _write_static_table(resource, table_name)
-                    _write_item(table_name, pipe_item.item)
+                    _write_item(table_name, resource.name, pipe_item.item)
                 else:
                     # get partial table from table template
                     if resource._table_name_hint_fun:
                         if isinstance(pipe_item.item, List):
                             for item in pipe_item.item:
-                                _write_dynamic_table(resource, item)
+                                table_name = resource._table_name_hint_fun(item)
+                                _write_dynamic_table(table_name, resource, item)
                         else:
-                            _write_dynamic_table(resource, pipe_item.item)
+                            table_name = resource._table_name_hint_fun(pipe_item.item)
+                            _write_dynamic_table(table_name, resource, pipe_item.item)
                     else:
                         # write item belonging to table with static name
                         table_name = resource.table_name
                         _write_static_table(resource, table_name)
-                        _write_item(table_name, pipe_item.item)
+                        _write_item(table_name, resource.name, pipe_item.item)
+
+            # find defined resources that did not yield any pipeitems and create empty jobs for them
+            tables_by_resources = utils.group_tables_by_resource(schema.tables)
```

Comment:
I think we can safely use `data_tables`. this eliminates `_dlt` tables and incomplete tables that were never created in the destination